main.go

```diff
@@ -302,15 +344,221 @@ func createPackagesGz(config conf, distro, section, arch string) error {
 			if i != (len(dirList) - 1) {
 				packBuf.WriteString("\n\n")
 			}
-			gzOut.Write(packBuf.Bytes())
+			writer.Write(packBuf.Bytes())
 			f = nil
 		}
 	}
 
-	gzOut.Flush()
 	return nil
 }
 
+func createRelease(config conf, distro string) error {
+
+	workingDirectory := filepath.Join(config.RootRepoPath, "dists", distro)
+
+	outfile, err := os.Create(filepath.Join(workingDirectory, "Release"))
+	if err != nil {
+		return fmt.Errorf("failed to create Release: %s", err)
+	}
+	defer outfile.Close()
+	
+	current_time := Now().UTC()
+    fmt.Fprintf(outfile, "Suite: %s\n", distro)
+    fmt.Fprintf(outfile, "Codename: %s\n", distro)
+    fmt.Fprintf(outfile, "Components: %s\n", strings.Join(config.Sections, " "))
+    fmt.Fprintf(outfile, "Architectures: %s\n", strings.Join(config.SupportArch, " "))
+    fmt.Fprintf(outfile, "Date: %s\n", current_time.Format("Mon, 02 Jan 2006 15:04:05 UTC"))
+
+	var md5Sums strings.Builder
+	var sha1Sums strings.Builder
+	var sha256Sums strings.Builder
+
+	err = filepath.Walk(workingDirectory, func(path string, file os.FileInfo, err error) error {
+	    if err != nil {
+	        return err
+	    }
+
+    	if strings.HasSuffix(path, "Packages.gz") || strings.HasSuffix(path, "Packages") {
+
+    		var relPath string
+    		relPath, err := filepath.Rel(workingDirectory, path)
+    		spath := filepath.ToSlash(relPath)
+    		f, err := os.Open(path)
+    		var (
+    			md5hash    = md5.New()
+    			sha1hash   = sha1.New()
+    			sha256hash = sha256.New()
+    		)
+    		if _, err = io.Copy(io.MultiWriter(md5hash, sha1hash, sha256hash), f); err != nil {
+    			log.Println("error with the md5 hash: ", err)
+    		}
+    		fmt.Fprintf(&md5Sums, " %s %d %s\n",
+    			hex.EncodeToString(md5hash.Sum(nil)),
+    			file.Size(), spath)
+
+    		if _, err = io.Copy(sha1hash, f); err != nil {
+    			log.Println("error with the sha1 hash: ", err)
```

Comment:
should this error stop everything? is it important enough to return here? i'm super foggy on the package stuff so i'll defer to you :)